src/mesh_lib/mod.rs

```diff
@@ -31,18 +28,20 @@ pub struct NodeConfig {
     // between `Node`s
     pub usart: Usart,
 
-    // The hardware abstraction library timer, which will be used for
-    // counting of milliseconds.
-    pub millis_timer: TC0,
+    // A closure for calculating millis
+    pub millis_fn_ptr: fn() -> ms,
```

Comment:
Maybe better:

```
trait PlatformTime {
  fn get_millis() -> ms;
}
```

mm?

Reply:
Maybe. and how it shall be organised within the struct? get_millis: impl PlatformTime ?

Reply:
Do not know, how to do it easier, and more flexible. Throw this PlatformTime trait implemented structure by the reference trough all the project, or make static reserved memory allocated place, and save initialized timer there, to let use the timer from statically allocated place.
![2023-12-11-043605_1324x641_scrot](https://github.com/boshtannik/NanoMesh/assets/22011895/e6f7264e-b511-46fb-84d5-eb6904e4dd34)

Reply:
@berkus, I have made some updates to use PlatformTime trait, instead function pointer. Made it as mothod, not as static associated function. Look please trough, tell what could be odd in this code, and could be simplified?

Reply:
Not actual anymore

Reply:
@berkus . Long story short. I'am merging with no review?